hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/validators/block/TransactionRecordParityValidator.java

```diff
@@ -0,0 +1,179 @@
+import com.hedera.hapi.node.transaction.TransactionRecord;
+import com.hedera.node.app.hapi.utils.forensics.DifferingEntries;
+import com.hedera.node.app.hapi.utils.forensics.RecordStreamEntry;
+import com.hedera.node.app.hapi.utils.forensics.TransactionParts;
+import com.hedera.node.app.state.SingleTransactionRecord;
+import com.hedera.services.bdd.junit.support.BlockStreamAccess;
+import com.hedera.services.bdd.junit.support.BlockStreamValidator;
+import com.hedera.services.bdd.junit.support.RecordStreamAccess;
+import com.hedera.services.bdd.junit.support.translators.BlockTransactionalUnitTranslator;
+import com.hedera.services.bdd.junit.support.translators.BlockUnitSplit;
+import com.hedera.services.bdd.spec.HapiSpec;
+import com.hedera.services.bdd.utils.RcDiff;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A validator that asserts the block stream contains all information previously exported in the record stream
+ * by translating the block stream into transaction records and comparing them to the expected records.
+ */
+public class TransactionRecordParityValidator implements BlockStreamValidator {
+    private static final int MAX_DIFFS_TO_REPORT = 10;
+    private static final int DIFF_INTERVAL_SECONDS = 300;
+    private static final Logger logger = LogManager.getLogger(TransactionRecordParityValidator.class);
+
+    private final BlockUnitSplit blockUnitSplit = new BlockUnitSplit();
+    private final BlockTransactionalUnitTranslator translator = new BlockTransactionalUnitTranslator();
+
+    public static final Factory FACTORY = new Factory() {
+        @NonNull
+        @Override
+        public TransactionRecordParityValidator create(@NonNull final HapiSpec spec) {
+            return new TransactionRecordParityValidator();
+        }
+
+        @Override
+        public boolean appliesTo(@NonNull final HapiSpec spec) {
+            requireNonNull(spec);
+            // Embedded networks don't have saved states or a Merkle tree to validate hashes against
+            return spec.targetNetworkOrThrow().type() == SUBPROCESS_NETWORK;
+        }
+    };
+
+    public static void main(String[] args) throws IOException {
+        final var input =
+                "/Users/michaeltinker/AlsoDev/hedera-services/hedera-node/test-clients/build/hapi-test/node0/data/block-streams/block-0.0.3/";
+        final var blocks = BlockStreamAccess.BLOCK_STREAM_ACCESS.readBlocks(Paths.get(input));
+        final var loc =
+                "/Users/michaeltinker/AlsoDev/hedera-services/hedera-node/test-clients/build/hapi-test/node0/data/recordStreams/record0.0.3";
+        final var records = RecordStreamAccess.RECORD_STREAM_ACCESS.readStreamDataFrom(loc, "sidecar");
```

Comment:
`michaeltinker`, Intentional?

Reply:
The main method is used for local testing.

Reply:
Right, just wondered if we wanted a placeholder other than `michaeltinker` 😂

Reply:
This could be done with no placeholder as follows:
```java
        final Path nodeZeroDataFolder =
                WorkingDirUtils.workingDirFor(0 "hapi").resolve("data").toAbsolutePath().normalize().toString();
        final String blockFolder = nodeZeroDataFolder.resolve("block-streams/block-0.0.3").toString();
        final String recordFolder = nodeZeroDataFolder.resolve("recordStreams/record0.0.3").toString();
```
Then use the two strings to get `blocks` and `records`.